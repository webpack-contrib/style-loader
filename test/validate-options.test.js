import loader from '../src/index';

it('validate options', () => {
  const validate = (options) =>
    loader.pitch.call(
      Object.assign(
        {},
        {
          query: options,
          loaders: [],
          remainingRequest: 'file.css',
          currentRequest: 'file.css',
          async: () => (error) => {
            if (error) {
              throw error;
            }
          },
        }
      ),
      'file.css'
    );

  expect(() => validate({ injectType: 'styleTag' })).not.toThrow();
  expect(() => validate({ injectType: 'singletonStyleTag' })).not.toThrow();
  expect(() => validate({ injectType: 'lazyStyleTag' })).not.toThrow();
  expect(() => validate({ injectType: 'lazySingletonStyleTag' })).not.toThrow();
  expect(() => validate({ injectType: 'linkTag' })).not.toThrow();
  expect(() =>
    validate({ injectType: 'unknown' })
  ).toThrowErrorMatchingSnapshot();

  expect(() => validate({ attributes: {} })).not.toThrow();
  expect(() => validate({ attributes: { id: 'id' } })).not.toThrow();
  expect(() => validate({ attributes: true })).toThrowErrorMatchingSnapshot();

  expect(() => validate({ insertAt: 'top' })).not.toThrow();
  expect(() =>
    validate({
      insertAt: {
        before: '#id',
      },
    })
  ).not.toThrow();
  expect(() => validate({ insertAt: true })).toThrowErrorMatchingSnapshot();

  expect(() =>
    // eslint-disable-next-line no-undef
    validate({ insertInto: () => document.querySelector('#root') })
  ).not.toThrow();
  expect(() => validate({ insertInto: 'test' })).not.toThrow();
  expect(() => validate({ insertInto: true })).toThrowErrorMatchingSnapshot();

  expect(() => validate({ unknown: 'unknown' })).toThrowErrorMatchingSnapshot();
});
