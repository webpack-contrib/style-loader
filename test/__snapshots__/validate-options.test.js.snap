// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate options should throw an error on the "attributes" option with "true" value 1`] = `
"Invalid options object. Style Loader has been initialized using an options object that does not match the API schema.
 - options.attributes should be an object:
   object { â€¦ }
   -> Adds custom attributes to tag.
   -> Read more at https://github.com/webpack-contrib/style-loader#attributes"
`;

exports[`validate options should throw an error on the "esModule" option with "true" value 1`] = `
"Invalid options object. Style Loader has been initialized using an options object that does not match the API schema.
 - options.esModule should be a boolean.
   -> Use the ES modules syntax.
   -> Read more at https://github.com/webpack-contrib/css-loader#esmodule"
`;

exports[`validate options should throw an error on the "injectType" option with "unknown" value 1`] = `
"Invalid options object. Style Loader has been initialized using an options object that does not match the API schema.
 - options.injectType should be one of these:
   \\"styleTag\\" | \\"singletonStyleTag\\" | \\"autoStyleTag\\" | \\"lazyStyleTag\\" | \\"lazySingletonStyleTag\\" | \\"lazyAutoStyleTag\\" | \\"linkTag\\"
   -> Allows to setup how styles will be injected into DOM.
   -> Read more at https://github.com/webpack-contrib/style-loader#injecttype"
`;

exports[`validate options should throw an error on the "insert" option with "true" value 1`] = `
"Invalid options object. Style Loader has been initialized using an options object that does not match the API schema.
 - options.insert should be one of these:
   string | function
   -> Inserts \`<style>\`/\`<link>\` at the given position.
   -> Read more at https://github.com/webpack-contrib/style-loader#insert
   Details:
    * options.insert should be a string.
    * options.insert should be an instance of function."
`;

exports[`validate options should throw an error on the "styleTagTransform" option with "[]" value 1`] = `
"Invalid options object. Style Loader has been initialized using an options object that does not match the API schema.
 - options.styleTagTransform should be one of these:
   string | function
   -> Transform tag and css when insert 'style' tag into the DOM
   -> Read more at https://github.com/webpack-contrib/style-loader#styleTagTransform
   Details:
    * options.styleTagTransform should be a string.
    * options.styleTagTransform should be an instance of function."
`;

exports[`validate options should throw an error on the "styleTagTransform" option with "true" value 1`] = `
"Invalid options object. Style Loader has been initialized using an options object that does not match the API schema.
 - options.styleTagTransform should be one of these:
   string | function
   -> Transform tag and css when insert 'style' tag into the DOM
   -> Read more at https://github.com/webpack-contrib/style-loader#styleTagTransform
   Details:
    * options.styleTagTransform should be a string.
    * options.styleTagTransform should be an instance of function."
`;

exports[`validate options should throw an error on the "unknown" option with "/test/" value 1`] = `
"Invalid options object. Style Loader has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'unknown'. These properties are valid:
   object { injectType?, attributes?, insert?, base?, esModule?, styleTagTransform? }"
`;

exports[`validate options should throw an error on the "unknown" option with "[]" value 1`] = `
"Invalid options object. Style Loader has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'unknown'. These properties are valid:
   object { injectType?, attributes?, insert?, base?, esModule?, styleTagTransform? }"
`;

exports[`validate options should throw an error on the "unknown" option with "{"foo":"bar"}" value 1`] = `
"Invalid options object. Style Loader has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'unknown'. These properties are valid:
   object { injectType?, attributes?, insert?, base?, esModule?, styleTagTransform? }"
`;

exports[`validate options should throw an error on the "unknown" option with "{}" value 1`] = `
"Invalid options object. Style Loader has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'unknown'. These properties are valid:
   object { injectType?, attributes?, insert?, base?, esModule?, styleTagTransform? }"
`;

exports[`validate options should throw an error on the "unknown" option with "1" value 1`] = `
"Invalid options object. Style Loader has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'unknown'. These properties are valid:
   object { injectType?, attributes?, insert?, base?, esModule?, styleTagTransform? }"
`;

exports[`validate options should throw an error on the "unknown" option with "false" value 1`] = `
"Invalid options object. Style Loader has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'unknown'. These properties are valid:
   object { injectType?, attributes?, insert?, base?, esModule?, styleTagTransform? }"
`;

exports[`validate options should throw an error on the "unknown" option with "test" value 1`] = `
"Invalid options object. Style Loader has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'unknown'. These properties are valid:
   object { injectType?, attributes?, insert?, base?, esModule?, styleTagTransform? }"
`;

exports[`validate options should throw an error on the "unknown" option with "true" value 1`] = `
"Invalid options object. Style Loader has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'unknown'. These properties are valid:
   object { injectType?, attributes?, insert?, base?, esModule?, styleTagTransform? }"
`;
